src/goto-instrument/cover_instrument_condition.cpp

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************\
+
+Module: Coverage Instrumentation
+
+Author: Daniel Kroening
+
+\*******************************************************************/
+
+/// \file
+/// Coverage Instrumentation for Conditions
+
+#include "cover_instrument.h"
+
+#include "cover_util.h"
+
+void cover_condition_instrumentert::instrument(
+  goto_programt &goto_program,
+  goto_programt::targett &i_it,
+  const cover_basic_blockst &basic_blocks) const
+{
+  if(
+    i_it->is_assert() &&
+    i_it->source_location.get_property_class() != property_class)
```

Comment:
At first reading I thought this would consider "location", "mcdc", and so on different classes, but it doesn't. Therefore suggest factoring this out into a function named `is_non_coverage_assertion` or something, and explain there that all coverage assertions are of the same class.

Reply:
Factored out. I also don't think that there are many use cases where replacing assert by skip makes sense.